src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -1155,38 +1155,47 @@ public async Task<OrganizationUser> AcceptUserAsync(string orgIdentifier, User u
                 }
             }
 
-            ICollection<Policy> orgPolicies = null;
-            ICollection<Policy> userPolicies = null;
-            async Task<bool> hasPolicyAsync(PolicyType policyType, bool useUserPolicies = false)
-            {
-                var policies = useUserPolicies ? 
-                    userPolicies = userPolicies ?? await _policyRepository.GetManyByUserIdAsync(user.Id) : 
-                    orgPolicies = orgPolicies ?? await _policyRepository.GetManyByOrganizationIdAsync(orgUser.OrganizationId);
-                
-                return policies.Any(p => p.Type == policyType && p.Enabled);
-            }
-            var userOrgs = await _organizationUserRepository.GetManyByUserAsync(user.Id);
-            if (userOrgs.Any(ou => ou.OrganizationId != orgUser.OrganizationId && ou.Status != OrganizationUserStatusType.Invited))
-            {   
-                if (await hasPolicyAsync(PolicyType.SingleOrg))
-                {
-                    throw new BadRequestException("You may not join this organization until you leave or remove " +
-                        "all other organizations.");
-                }
-                if (await hasPolicyAsync(PolicyType.SingleOrg, true))
-                {
-                    throw new BadRequestException("You cannot join this organization because you are a member of " + 
-                        "an organization which forbids it");
-                }
+            bool notExempt(OrganizationUser organizationUser)
+            {
+                return organizationUser.Type != OrganizationUserType.Owner &&
+                        organizationUser.Type != OrganizationUserType.Admin;
             }
 
-            if (!await userService.TwoFactorIsEnabledAsync(user))
+            var allOrgUsers = await _organizationUserRepository.GetManyByUserAsync(user.Id);
+
+            // Enforce Single Organization Policy of organization user is trying to join
+            var thisSingleOrgPolicy = await _policyRepository.GetByOrganizationIdTypeAsync(orgUser.OrganizationId, PolicyType.SingleOrg);
+            if (thisSingleOrgPolicy != null &&
+                thisSingleOrgPolicy.Enabled &&
+                notExempt(orgUser) &&
+                allOrgUsers.Count(ou => ou.OrganizationId != orgUser.OrganizationId) > 0)
```

Comment:
Change to `.Any()` vs. `.Count() > 0`; best practice is to avoid the extra loop cycles, Any() exits and returns `true` the second it finds a match, vs. Count() which still iterates through all in the collection with an accumulator before returning a final count.